### ISSUE TITLE
ios tests

### ISSUE THREAD

**Issue by tofutim** [oysteinkrog/SQLite.Net-PCL#253]
Has anyone done testing on iOS? I find that a number of tests just hang...

**Comment by tofutim** [comment]
For example in the AyncTests, looks like it hangs at await conn.CreateTableAsync<Customer>(); It does start running it, and get past 

            var result = new CreateTablesResult();
                var conn = GetConnection();
                using (conn.Lock())
                {
                    foreach (var type in types)
                    {
                        var aResult = conn.CreateTable(type);
                        result.Results[type] = aResult;
                    }
                }
                return result;

to return result, and then the Debugger seems to get lost and that's it

**Comment by tofutim** [comment]
ok, it also fails with something simple like

	[TestFixture]
	public class awaitTest
	{
		[Test]
		public async Task Pass ()
		{
			await Task.Delay (1000);
			Assert.True (true);
		}
	}

See https://bugzilla.xamarin.com/show_bug.cgi?id=15104 - for iOS, 

"Tests that cannot be executed from the main thread can use the [Timeout]
attribute (e.g. with Int32.MaxValue) to have the nunit framework run it into a
separate thread.

That keeps the default Touch.Unit runner behaviour of using the main (UI)
thread - and let you mark any test that you need to execute on a different
thread."

So for all the async tests, we should add [Timeout (Int32.MaxValue)], e.g., 

	[TestFixture]
	public class awaitTest
	{
		[Test]
		[Timeout (Int32.MaxValue)]
		public async Task Pass ()
		{
			await Task.Delay (1000);
			Assert.True (true);
		}
	}

TL;DR Tests in iOS are run on the main thread, so the UI gets frozen, use [Timeout (Int32.MaxValue)] as needed

**Comment by tofutim** [comment]
That definitely gets it to go - and the UI is no longer frozen. May want to make this only an iOS or Android test flag. After this is run, a new error is introduced, after the first three async runs

![image](https://cloud.githubusercontent.com/assets/157766/11162586/d4c2dd52-8a5d-11e5-9ed9-764ecbe5bd59.png)

**Comment by oysteinkrog** [comment]
Hi,
Yeah I'm aware of some of these problems, to be honest I've been kind of waiting for NUnit 3.0 and the new Xamarin support there, in the hope that it would support async tests better.
If there is a good way to fix this now though I am ofcourse not opposed to it:)
